src/lm_buddy/jobs/_entrypoints/prometheus.py

```diff
@@ -149,32 +150,50 @@ def run_eval(
             file.write(json.dumps(result) + "\n")
 
     # convert plain json dataset in HF format
-    output_hf_name = str(Path(config.evaluation.output_folder) / "hf" / tracking_name)
+    output_dataset_path = Path(config.evaluation.output_folder) / "hf" / tracking_name
     ds = load_dataset("json", data_files=str(output_fname), split="train")
-    ds.save_to_disk(output_hf_name)
+    ds.save_to_disk(output_dataset_path)
 
-    return str(output_hf_name)
+    return output_dataset_path
 
 
-def run_prometheus(config: PrometheusJobConfig, artifact_loader: ArtifactLoader):
-    # instantiate OpenAI client to speak with the vLLM endpoint
+def run_prometheus(
+    config: PrometheusJobConfig,
+    artifact_loader: ArtifactLoader,
+) -> EvaluationResult:
+    # Instantiate OpenAI client to speak with the vLLM endpoint
     client = OpenAI(base_url=config.prometheus.inference.base_url)
 
-    # Register a dataset file artifact if tracking is enabled
+    # Run eval and store output in local filename
     if config.tracking:
-        with wandb_init_from_config(config.tracking, job_type=LMBuddyJobType.EVALUATION):
-            # run eval and store output in local filename
-            output_dataset_name = run_eval(config, artifact_loader, client)
-
-            # store HF dataset as a directory artifact
-            artifact = build_directory_artifact(
-                dir_path=output_dataset_name,
-                artifact_name=config.tracking.name,
+        with wandb_init_from_config(config.tracking, job_type=LMBuddyJobType.EVALUATION) as run:
+            output_dataset_path = run_eval(config, artifact_loader, client)
+
+            # Create a directory artifact for the HF dataset
+            dataset_artifact = build_directory_artifact(
```

Comment:
suggestion: might be helpful to add types to any variable assignments where we generate artifacts or data outputs so we can easily tell what's generated